### ISSUE TITLE
Confusing doctor output if dependency is out of date

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#6285]
I just ran `flutter doctor`, which has a bit of a confusing output. I had an out-of-date dependency, but doctor reported "checkmark" for iOS toolchain.

Suggestion: report a "not checkmark" for a category, when a dependency is incorrect/out of date.

output:

```
[✓] iOS toolchain - develop for iOS devices (Xcode 8.0)
    • XCode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.0, Build version 8A218a
    x libimobiledevice is incompatible with the installed XCode version. To update, run:
      brew uninstall libimobiledevice
      brew install --HEAD libimobiledevice
```

Suggestion:

```
[X] iOS toolchain - develop for iOS devices (Xcode 8.0)
    • XCode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.0, Build version 8A218a
    x libimobiledevice is incompatible with the installed XCode version. To update, run:
      brew uninstall libimobiledevice
      brew install --HEAD libimobiledevice
```

**Comment by sethladd** [comment]
cc @danrubel @cbracken

**Comment by cbracken** [comment]
On it -- thanks for spotting!

**Comment by cbracken** [comment]
Fixed by #6289.
